system-addon/common/ShortURL.jsm

```diff
@@ -15,18 +18,34 @@ Components.utils.importGlobalProperties(["URL"]);
  *         {str} link.title (optional) - The title of the link
  * @return {str}   A short url
  */
-this.shortURL = function shortURL(link) {
-  if (!link.url && !link.hostname) {
-    return "";
+this.ShortURL = class ShortURL {
+  constructor() {
+    this.IDNService = Cc["@mozilla.org/network/idn-service;1"].getService(Ci.nsIIDNService);
```

Comment:
Unless we're adding other methods, probably simpler to just leave this `jsm` as returning the single `function` instead of a `class`. You can probably have a lazy service getter, e.g., https://github.com/mozilla/activity-stream/blob/14c051f5bbfcda584532bf4ecc7c276973a15300/system-addon/lib/Screenshots.jsm#L17-L18

Then you can just use the global `IDNService` without storing it on an instance.